parser-typechecker/src/Unison/Codebase/Branch.hs

```diff
@@ -338,10 +357,15 @@ deepEdits' = go id where
     f :: (NameSegment, Branch m) -> Map Name (EditHash, m Patch)
     f (c, b) =  go (addPrefix . Name.cons c) (head b)
 
--- Discards the history of a Branch0's children, recursively
+-- | Discards the history of a Branch0's children, recursively
 discardHistory0 :: Applicative m => Branch0 m -> Branch0 m
 discardHistory0 = over children (fmap tweak) where
-  tweak b = cons (discardHistory0 (head b)) empty
+  tweak b = one (discardHistory0 (head b))
```

Comment:
Previously this always appended onto an `empty`; but according to a [slack convo](https://unisoncomputing.slack.com/archives/CF8D3DCP3/p1637682445316200) we want to _start_ the branch with the first changes rather than having an empty branch there.